tests/test_context.py

```diff
@@ -21,9 +21,24 @@ def test_context_with_path(self):
             ignore_dependencies=sentinel.ignore_dependencies,
         )
 
-        sentinel.project_path = "project_path/to/sceptre"
+        sentinel.project_path = f"{getcwd()}/project_path/to/sceptre"
```

Comment:
Why did you make this change? If this test started failing when you updated the way the path is resolved, then your change cannot work. It will fail in a lot of places where people currently have their CI/CD set to use `--dir project_path/to/sceptre`

Reply:
Sorry, I think this is a misunderstanding on my part of how that test was working.

Maybe it should instead be:
```
expected = f"{getcwd()}/project_path/to/sceptre"
assert self.context.project_path.replace(path.sep, "/") == expected
```

The `--dir` flag does work with the change as it's coercing any given project_path into an absolute path when SceptreContext is initialised

Reply:
I'll need to do some experimentation on this branch, I think. My own projects use `--dir` quite often.